### ISSUE TITLE
A new logging method for auxillary module is needed.

### ISSUE THREAD

**Issue by Ryuuuuu** [rapid7/metasploit-framework#17852]
## Summary

Currently, the format of auxiliary module logging varies from module to module, and when a module is successful, it mostly outputs logging via print_good.

However, since the module for detecting vulnerabilities and the module for searching simple information exist together in the auxiliary module, it is difficult to determine whether or not the target is vulnerable at  a glance

It would be good to divide the logging method to determine at a glance whether the target is vulnerable.

It will not be easy because there are many existing modules, but it would be nice to have such a function.

### Basic example

It would be good to divide the logging method to determine at a glance whether the target is vulnerable.

like print_vuln and print_info

### Motivation

The use of auxiliary modules is likely to be much easier.

**Comment by adfoster-r7** [comment]
> the format of auxiliary module logging varies from module to module,

Adding to this; Auxiliary scanner modules that run against multiple hosts in parallel can often be very quiet by default, even when run against one host - which can also be confusing to end users.

**Comment by saurabhmj11** [comment]
One way to achieve this could be to define specific logging functions for vulnerability detection and information gathering. For example, you could define a log_vuln function to log vulnerability detection and a log_info function to log information gathering.

Here is an example of how this could look in code:

python
Copy code
import logging

# Create a logger
logger = logging.getLogger(__name__)

# Define logging functions
def log_vuln(msg):
    logger.warning("[VULN] %s", msg)

def log_info(msg):
    logger.info("[INFO] %s", msg)
You can then use these functions throughout your modules to log vulnerability detection and information gathering messages. This will help make it easier to determine if a target is vulnerable or not at a glance.

Note that you can customize the logging format and output destination to suit your needs using Python's logging module.
